File: KITE-Janus-Test/src/main/java/io/cosmosoftware/kite/janus/steps/ScreenshotStep.java
package io.cosmosoftware.kite.janus.steps;

import io.cosmosoftware.kite.janus.Scenario;
import io.cosmosoftware.kite.report.Reporter;
import io.cosmosoftware.kite.steps.TestStep;
import org.openqa.selenium.WebDriver;

import static io.cosmosoftware.kite.util.ReportUtils.saveScreenshotPNG;
import static io.cosmosoftware.kite.util.ReportUtils.timestamp;

public class ScreenshotStep extends TestStep {

  public ScreenshotStep(WebDriver webDriver, Scenario scenario) {
    super(webDriver);
    this.setName(getName() + "_" + scenario.getName());
  }

  public ScreenshotStep(WebDriver webDriver) {
    super(webDriver);
  }
  
  
  @Override
  public String stepDescription() {
    return "Get a screenshot";
  }
  
  @Override
  protected void step() {
    Reporter.getInstance().screenshotAttachment(report,
      "ScreenshotStep_" + timestamp(), saveScreenshotPNG(webDriver));
  }
}

File: KITE-Janus-Test/js/steps/ScreenshotStep.js
const {TestUtils, TestStep} = require('kite-common');

/**
 * Class: ScreenshotStep
 * Extends: TestStep
 * Description:
 */
class ScreenshotStep extends TestStep {
  constructor(kiteBaseTest) {
    super();
    this.driver = kiteBaseTest.driver;

    // Test reporter if you want to add attachment(s)
    this.testReporter = kiteBaseTest.reporter;
  }

  stepDescription() {
    return 'Get a screenshot';
  }

  async step() {
    let screenshot = await TestUtils.takeScreenshot(this.driver);
    this.testReporter.screenshotAttachment(this.report, "Screenshot step", screenshot); 
  }
}

module.exports = ScreenshotStep;